### DISCUSSION TITLE
Baked SDF for ray-traced shaders

### DISCUSSION THREAD

**Discussion by ElSargo** [IrisShaders/Iris#1371] [Ideas]
I was looking into ue5 and found they were using raymarching SDFs for GI.
![raymarching-0](https://user-images.githubusercontent.com/103979764/163917167-f9654d2a-7e3a-47e2-914d-7a8732f78e30.png)
  If distance information could be baked in perhaps a voxel format, sphere tracing could be used to greatly speed up trace. Also since most geometry is just blocks, the volume could be baked at a very low resolution. Distance info would have other advantages, even for non-rt shaders such as fast non-screen space AO and cheap soft shadows. 
![gfx03](https://user-images.githubusercontent.com/103979764/163917648-734600e3-03d3-4119-b073-9a2a6b98d961.png)
I don't know how to do mc mods but I have written raymarching shaders before and think this would be a great addition to this already amazing mod!

**Comment by qb20nh** [comment]
I think this is something each shader should choose to implement, and not come from iris. We would have to re-calculate the bake information every time block is broken/placed/moved/model was changed and what about entities...

**Reply by ElSargo** [reply]
Yeah, it would be ideal if shaders could implement this themselves, but I don't think that it's feasible and it would be much better suited for a   mod. I was thinking that it would be similar to the shadow map in that the pack could apply some settings like on/off, resolution, etc. 

As for entities, I think that they should just be omitted from the bake as they don't really influence the lighting much and aren't considered in GI anyway.

Also if a max distance was set on how far the bake would search, only cells within that radius would need their distance updated, preventing the entire world from being recalculated. I'm also fairly confident that a compute shader would demolish a small rebake in under a millisecond since it will probably be fairly high-end machines running the kinds of shaders that would use this info.
